execution_engine/src/shared/transform.rs

```diff
@@ -211,6 +212,7 @@ impl Transform {
                 }
             },
             Transform::Failure(error) => Err(error),
+            Transform::Delete => Ok(None),
```

Comment:
There is a comment in `tracking_copy/mod.rs` saying Transform shouldn't ever produce delete, so i'm confused on what `Transform::Delete` represents here, maybe Im confusing the ordering here???